engine/src/flutter/tools/mcp/lib/server.dart

```diff
@@ -0,0 +1,143 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async' show Completer;
+import 'dart:convert' show LineSplitter, utf8;
+import 'dart:io' show Process;
+
+import 'package:mcp_dart/mcp_dart.dart';
+import 'package:process_runner/process_runner.dart';
+
+McpServer makeServer({ProcessRunner? processRunner}) {
+  processRunner ??= ProcessRunner();
+  final McpServer server = McpServer(
+    const Implementation(name: 'engine_mcp', version: '1.0.0'),
+    options: const ServerOptions(
+      capabilities: ServerCapabilities(
+        resources: ServerCapabilitiesResources(),
+        tools: ServerCapabilitiesTools(),
+      ),
+    ),
+  );
+
+  server.tool('engine_build_help',
+      description: 'Get help for the building tool and a list of configs.',
+      callback: ({Map<String, dynamic>? args, RequestHandlerExtra? extra}) async {
+    try {
+      final List<String> arguments = ['./bin/et', 'build', '--help'];
+      final ProcessRunnerResult result = await processRunner!.runProcess(arguments);
+      final String output = result.stdout;
+
+      return CallToolResult(
+        content: [
+          TextContent(
+            text: output,
+          ),
+        ],
+      );
+    } catch (ex) {
+      return CallToolResult.fromContent(content: [TextContent(text: ex.toString())], isError: true);
+    }
+  });
+
+  server.tool(
+    'engine_list_targets',
+    description: 'Lists build targets for a given config.', //
+    inputSchemaProperties: {
+      'config': {
+        'type': 'string',
+        'enum': ['host_debug_unopt', 'host_debug_unopt_arm64'],
```

Comment:
Is the goal to have this come dynamically at some point?

Reply:
I'm not sure.  It doesn't 100% follow this guidance anyway.  I think we'll have to use it and see where it seems to fail.  In theory it can query these with `engine_build_help`.  I'm not sure if we should add a `engine_get_configs` to be more apropos.